src/main/kotlin/org/rust/ide/utils/import/importUtils.kt

```diff
@@ -46,6 +47,7 @@ fun ImportCandidateBase.import(context: RsElement) {
             ((scope as? RsFunction)?.block ?: scope) as RsItemsOwner
         }
         containingFile is RsCodeFragment -> containingFile.importTarget
+        containingFile is RsFile && RsIncludeMacroIndex.getIncludedFrom(containingFile) != null -> containingFile
```

Comment:
I think it's ok for now but it potentially makes index lookup in write action that may be quite slow.
So probably, we need to extract insert location before write action invocation